src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.SemanticChangeProcessor.cs

```diff
@@ -183,13 +183,11 @@ private async Task EnqueueWorkItemAsync(Document thisDocument, ImmutableArray<Lo
                     {
                         Debug.Assert(location.IsInSource);
 
-                        var document = solution.GetDocument(location.SourceTree, projectId);
-                        if (document == null || thisDocument == document)
-                        {
+                        var documentId = solution.GetDocumentId(location.SourceTree, projectId);
+                        if (documentId == null || thisDocument.Id == documentId)
                             continue;
-                        }
 
-                        await _processor.EnqueueWorkItemAsync(document).ConfigureAwait(false);
+                        await _processor.EnqueueWorkItemAsync(solution.GetRequiredProject(documentId.ProjectId), documentId, document: null).ConfigureAwait(false);
```

Comment:
Do we want to have some comment here that we're trying to avoid grabbing the Document? I can imagine somebody else coming back to this and wondering why it wasn't written the original way.